pkg/pillar/types/newlogtypes.go

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2020 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package types
+
+import (
+	"time"
+)
+
+const (
+	// DevPrefix - device log file prefix string
+	DevPrefix = "dev.log."
+	// AppPrefix - app log file prefix string
+	AppPrefix = "app."
+	// AppSuffix - app log file suffix string, the appuuid is between the AppPrefix and AppSuffix
+	AppSuffix = ".log."
+)
+
+type logfileMetrics struct {
+	// from newlogd
+	NumGZipFilesSent  uint64 // total gzip files uploaded, device
+	NumGZipBytesWrite uint64 // total gzip log in bytes, device
+	NumBytesWrite     uint64 // total log bytes write to file before gzip, device
+	NumGzipFileInDir  uint32 // current number of gzip files remain, device
+	NumInputEvent     uint64 // total event input from log source, device include kmessages
+	// from loguploader
+	NumGZipFileRetry      uint64    // total gzip file upload retries, device
+	NumGZipFileDrop       uint32    // total gzip file upload 4xx failure and kept on device
+	RecentUploadTimestamp time.Time // uploaded to cloud the most recent log timestamp
+	LastGZipFileSendTime  time.Time // last upload gzip file time, device
+}
+
+type serverStats struct {
+	CurrCPULoadPCT  float32 // newlog server CPU percentage usage
+	AvgCPULoadPCT   float32 // newlog server CPU avarage load percentage
+	CurrProcessMsec uint32  // newlog server process log duration in msec
+	AvgProcessMsec  uint32  // newlog server avg process log duration in msec
+}
+
+type cloudDelay struct {
+	MinUploadMsec  uint32 // min upload to cloud delay in msec
+	MaxUploadMsec  uint32 // max upload to cloud delay in msec
+	AvgUploadMsec  uint32 // avg upload to cloud delay in msec
+	CurrUploadMsec uint32 // current upload to cloud delay in msec
+}
+
+// NewlogMetrics - Metrics from newlogd and loguploader
+type NewlogMetrics struct {
+	// logupload signal to newlogd
+	FailedToSend      bool      // loguploader failed to send to cloud
```

Comment:
boolean and not a count of how many times it failed?

Reply:
there is a local variable keep track of how many times failed, this is for publication from loguploader to newlogd that it'
s currently in or not in 'FailedToSend' state.

Reply:
ok